### PR TITLE
docs: prepare docs for Ruby SDK release

### PR DESCRIPTION
**Pull request by jeffomatic**
This commit handles the low-hanging fruit for the Ruby SDK
release, including:

- adding "download" instructions in the SDK doc
- adding links to the bundled YARD API docs in the nav menu
- adding code-sample download links to each doc

Future commits will handle the harder work, including:

- A language switching mechanism for code snippets
- Ensuring that prose content does not include Java-specific
  verbiage

### CODE REVIEW

**Comment by jeffomatic** [comment]
PTAL

File: docs/core/build-applications/transactions.md [hunk 1]
```diff
@@ -1,232 +0,0 @@
-# Transactions Guide
```
Comment (dominic) [inline docs/core/build-applications/transactions.md]: Was this a deprecated doc?

Comment (jeffomatic) [inline docs/core/build-applications/transactions.md]: Yeah we forgot to delete it.

Comment (dominic) [inline docs/core/build-applications/transactions.md]: :shipit:

**Pull request merged**